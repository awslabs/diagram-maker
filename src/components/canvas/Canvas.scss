@import './patterns/DotPattern.scss';
@import './patterns/GridPattern.scss';

.dm-canvas {
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: #f2f3f3;
  pointer-events: none;
  user-select: none;
  z-index: 1;

  > svg {
    position: absolute;
    width: 100%;
    height: 100%;
  }
}
