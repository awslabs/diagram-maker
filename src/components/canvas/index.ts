export { default as Canvas, CanvasProps } from './Canvas';
