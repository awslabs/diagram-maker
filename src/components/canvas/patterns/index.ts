export { default as DotPattern, DotPatternProps } from './DotPattern';
export { default as GridPattern, GridPatternProps } from './GridPattern';
