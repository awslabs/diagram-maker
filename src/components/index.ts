export { Canvas, CanvasProps } from './Canvas';
export { default as ComposeView } from './common/ComposeView';
export {
  default as Connector,
  ConnectorProps,
  ConnectorType
} from './connector/Connector';
export {
  default as ContextMenu,
  ContextMenuProps
} from './contextMenu/ContextMenu';
export { Node, NodeProps, PotentialNode, PotentialNodeProps } from './node';
export { Panel, PanelProps } from './panel';
export { SelectionMarquee, SelectionMarqueeProps } from './selectionMarquee';
export { View, ViewProps } from './view/View';
export { getConnectedView, ConnectedViewOwnProps } from './view/ConnectedView';
export { Workspace, WorkspaceProps } from './Workspace';

export * from './edge/EdgeCurve';
export { default as Edge, EdgeProps } from './edge/Edge';
export { default as EdgeBadge, EdgeBadgeProps } from './edge/EdgeBadge';
export {
  default as PotentialEdge,
  PotentialEdgeProps
} from './edge/PotentialEdge';

export {
  default as GridPattern,
  GridPatternProps
} from './canvas/patterns/GridPattern';

export {
  default as DotPattern,
  DotPatternProps
} from './canvas/patterns/DotPattern';

export { render, destroy } from './renderUtils';
