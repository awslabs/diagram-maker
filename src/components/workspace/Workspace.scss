.dm-workspace {
  position: absolute;
  transform-origin: 0 0;
  overflow: hidden;
}
