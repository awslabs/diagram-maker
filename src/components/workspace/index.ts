export { default as Workspace, WorkspaceProps } from './Workspace';
