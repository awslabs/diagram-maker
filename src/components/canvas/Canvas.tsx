import * as classnames from 'classnames';
import * as Preact from 'preact';

import './Canvas.scss';

export interface CanvasProps {
  className?: string | string[];
}

const Canvas = (props: Preact.RenderableProps<CanvasProps>): JSX.Element => {
  const className = classnames('dm-canvas', props.className);

  return (
    <div className={className}>
        {props.children}
    </div>
  );
};

export default Canvas;
