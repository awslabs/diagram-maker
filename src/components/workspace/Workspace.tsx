import * as classnames from 'classnames';
import * as Preact from 'preact';

import Canvas from 'diagramMaker/components/canvas/Canvas';
import DotPattern from 'diagramMaker/components/canvas/patterns/DotPattern';
import { DiagramMakerComponentsType } from 'diagramMaker/service/ui/types';
import { Position, Size } from 'diagramMaker/state/types';

import './Workspace.scss';

export interface WorkspaceProps {
  canvasPattern?: JSX.Element;
  className?: string | string[];
  position: Position;
  scale: number;
  canvasSize: Size;
}

export const DEFAULT_CANVAS_PATTERN: JSX.Element = <DotPattern cellSize={10} radius={1} />;

const Workspace = (props: Preact.RenderableProps<WorkspaceProps>): JSX.Element => {
  const className = classnames('dm-workspace', props.className);
  const { height, width } = props.canvasSize;
  const { x, y } = props.position;
  const scale = props.scale;

  const transform = `translate3d(${x}px, ${y}px, 0) scale(${scale})`;

  const canvasPattern = props.canvasPattern || DEFAULT_CANVAS_PATTERN;

  return (
    <div
      className={className}
      style={{ height, width, transform }}
      data-type={DiagramMakerComponentsType.WORKSPACE}
      data-event-target={true}
      data-draggable={true}
      data-dropzone={true}
    >
        <Canvas>
          {canvasPattern}
        </Canvas>
        {props.children}
    </div>
  );
};

export default Workspace;
