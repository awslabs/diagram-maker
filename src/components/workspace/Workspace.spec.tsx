import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { DotPattern, GridPattern } from 'diagramMaker/components/canvas/patterns';
import { Workspace } from 'diagramMaker/components/workspace';

describe('Workspace', () => {
  it('renders canvas given dot pattern', () => {
    const dotPattern = <DotPattern cellSize={10} radius={1} />;

    const workspace = shallow(
            <Workspace
              className="workspace"
              canvasPattern={dotPattern}
              position={{ x:0, y:0 }}
              canvasSize={{ width:800, height:400 }}
              scale={1}
            />
    );

    expect(workspace).toMatchSnapshot();
  });

  it('renders canvas given grid pattern', () => {
    const gridPattern = <GridPattern cellSize={10} patternSize={100}/>;

    const workspace = shallow(
      <Workspace
        className="workspace"
        canvasPattern={gridPattern}
        position={{ x:0, y:0 }}
        canvasSize={{ width:800, height:400 }}
        scale={1}
      />
    );

    expect(workspace).toMatchSnapshot();
  });

  it('renders dot pattern as default', () => {
    const workspace = shallow(
      <Workspace
        className="workspace"
        position={{ x:0, y:0 }}
        canvasSize={{ width:800, height:400 }}
        scale={1}
      />
    );

    expect(workspace).toMatchSnapshot();
  });

});
