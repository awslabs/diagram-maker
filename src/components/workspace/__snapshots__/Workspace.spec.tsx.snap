// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace renders canvas given dot pattern 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-workspace workspace"
  style="height: 400px; width: 800px; transform: translate3d(0px, 0px, 0) scale(1);"
  data-type="DiagramMaker.Workspace"
  data-event-target={true}
  data-draggable={true}
  data-dropzone={true}
>
  <Canvas>
    <DotPattern
      cellSize={10}
      radius={1}
    >
    </DotPattern>
  </Canvas>
</div>

`;

exports[`Workspace renders canvas given grid pattern 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-workspace workspace"
  style="height: 400px; width: 800px; transform: translate3d(0px, 0px, 0) scale(1);"
  data-type="DiagramMaker.Workspace"
  data-event-target={true}
  data-draggable={true}
  data-dropzone={true}
>
  <Canvas>
    <GridPattern
      cellSize={10}
      patternSize={100}
    >
    </GridPattern>
  </Canvas>
</div>

`;

exports[`Workspace renders dot pattern as default 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-workspace workspace"
  style="height: 400px; width: 800px; transform: translate3d(0px, 0px, 0) scale(1);"
  data-type="DiagramMaker.Workspace"
  data-event-target={true}
  data-draggable={true}
  data-dropzone={true}
>
  <Canvas>
    <DotPattern
      cellSize={10}
      radius={1}
    >
    </DotPattern>
  </Canvas>
</div>

`;
